pkg/api/operators/apis/experiment/v1alpha2/types.go

```diff
@@ -0,0 +1,224 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	pb "github.com/kubeflow/katib/pkg/api"
+	trial "github.com/kubeflow/katib/pkg/api/operators/apis/trial/v1alpha2"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ExperimentSpec struct {
+	// List of hyperparameter configurations.
+	Parameters []ParameterSpec `json:"parameters,omitempty"`
+
+	// Describes the objective of the experiment.
+	Objective *ObjectiveSpec `json:"objective,omitempty"`
+
+	// Describes the suggestion algorithm.
+	Algorithm *AlgorithmSpec `json:"algorithm,omitempty"`
+
+	// Template for each run of the trial.
+	TrialTemplate *TrialTemplate `json:"trialTemplate,omitempty"`
+
+	// How many trials can be processed in parallel.
+	ParallelTrialCount int `json:"parallelTrialCount,omitempty"`
+
+	// Total number of trials to run.
+	MaxTrialCount      int `json:"maxTrialCount,omitempty"`
+
+	// TODO - figure out what to do with metric collectors
+	MetricsCollectorType string `json:"metricsCollectorSpec,omitempty"`
+
+	// TODO - Other fields, exact format is TBD. Will add these back during implementation.
+	// - NAS
+	// - Early stopping
+	// - Resume experiment
+}
+
+type ExperimentStatus struct {
+	// Represents time when the Experiment was acknowledged by the Experiment controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime         *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the Experiment was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime    *metav1.Time `json:"completionTime,omitempty"`
```

Comment:
May be more conventional to use `EndTime`?

Reply:
CompletionTime is more consistent with kubernetes native resources, for example BatchJobs: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/types.go#L156